### ISSUE TITLE
Running on mobile devices

### ISSUE THREAD

**Issue by LSQzzx** [ByteDance-Seed/Depth-Anything-3#63]
Is it possible to convert dav3 into coreml model (mlmodel, mlpackages) or executorch format (.pte) ? It seems that there are a lot of Python data structures (like dict,list...) mixed in the network, so I tried many different methods and all resulted in strange error messages.

**Comment by olokobayusuf** [comment]
@LSQzzx currently working on this! Across ONNX and CoreML, there's a `torch.nn.functional.interpolate` call which uses `cubic` mode (unsupported by both ONNXRuntime and CoreML). We're working on a patch for this [that works for ONNX using constant folding](https://github.com/muna-ai/muna-predictors/blob/f858d454ace13a84d814167b9bea36ccacabe1fb/depth-estimation/depth_anything_v3_.py#L82).

For CoreML, we're working on patching the interpolation somehow. [Join the convo](https://muna.ai/slack).

**Comment by LSQzzx** [comment]
> Is it possible to convert dav3 into coreml model (mlmodel, mlpackages) or executorch format (.pte) ? It seems that there are a lot of Python data structures (like dict,list...) mixed in the network, so I tried many different methods and all resulted in strange error messages.

I successfully compiled mlpackage. Welcome to my fork [Depth-Anything-3-for-CoreML](https://github.com/LSQzzx/Depth-Anything-3-for-CoreML)

**Comment by olokobayusuf** [comment]
[We did the same](https://gist.github.com/olokobayusuf/bebf8fe0bd766d51dcbb16e11b815926).

Try it yourself: [Install the Muna CLI](https://docs.muna.ai/introduction) then run [the compiled model](https://www.muna.ai/@yusuf/depth-anything-v3-metric-large-coreml) on a test image (assuming you have `test.jpg` in the same directory):
```sh
# Run Depth Anything v3 with CoreML
$ muna predict yusuf/depth-anything-v3-metric-large-coreml --image @test.jpg
```

In our benchmarking, the CoreML DAv3-large model runs in 300-400ms whereas ONNXRuntime takes over 2000ms (>5x slower). CoreML FTW.

**Comment by Ashwani564** [comment]
Your deployed model at Muna has following error: Cannot create @yusuf/depth-anything-v3-metric-large-coreml prediction because predictor does not have an active implementation for client 'wasm32'.

**Comment by olokobayusuf** [comment]
> Your deployed model at Muna has following error: Cannot create @yusuf/depth-anything-v3-metric-large-coreml prediction because predictor does not have an active implementation for client 'wasm32'.

This predictor was compiled to only use CoreML, which is only available on Apple Silicon (iOS and macOS). [See this line](https://gist.github.com/olokobayusuf/bebf8fe0bd766d51dcbb16e11b815926#file-depth_anything_v3_coreml-py-L106).

The [original implementation](https://github.com/muna-ai/muna-predictors/blob/main/depth-estimation/depth_anything_v3_.py) we compiled runs everywhere: [`@bytedance/depth-anything-v3-metric-large`](https://www.muna.ai/@bytedance/depth-anything-v3-metric-large).
